### DISCUSSION TITLE
IP NETS external firewall uri rules to block

### DISCUSSION THREAD

**Discussion by docbt** [DigitalRuby/IPBan#164] [Ideas]
Hi!

Is it it possible to use external IP nets lists as external uri rules to block? Or only IPs?

I would like to add https://www.spamhaus.org/drop/, which uses IPv4 and IPv6 ip nets (e.g. 1.10.16.0/20 or 2a00:4c80::/29)

Thanks in advance!
Greetings Andy

**Comment by jjxtra** [comment]
Should work just fine. There is a firewall uri rules property in the config you can use.

**Comment by docbt** [comment]
Hi Jeff!
Thanks for your answer.
Firewall uri rules work for lists with ip addresses, but not with the Spamhaus lists.
I added 3 lists with IP adresses, but the Spamhaus lists don´t work...
I think the problem is the **;** at the enf of each line. Could you change your code to ignore ; please?
Greetings
Andy

A must-have list in my opinion:
"The Spamhaus DROP (Don't Route Or Peer) lists are advisory "drop all traffic" lists, consisting of netblocks that are "hijacked" or leased by professional spam or cyber-crime operations (used for dissemination of malware, trojan downloaders, botnet controllers). "

Example:
; Spamhaus DROP List 2021/12/06 - (c) 2021 The Spamhaus Project
; https://www.spamhaus.org/drop/drop.txt
; Last-Modified: Tue, 30 Nov 2021 13:22:28 GMT
; Expires: Mon, 06 Dec 2021 07:17:39 GMT
1.10.16.0/20 ; SBL256894
1.19.0.0/16 ; SBL434604
1.32.128.0/18 ; SBL286275
2.56.192.0/22 ; SBL459831
2.58.176.0/23 ; SBL494020
2.58.178.0/23 ; SBL494021
2.59.200.0/22 ; SBL463003

**Comment by jjxtra** [comment]
Hmm, I think the ; is throwing it off, I'll have to add that as a comment character

**Comment by jjxtra** [comment]
I've refreshed the 1.6.1 release to add the ; as a comment delimiter, let me know if it works any better for you
